datastore/db/bytebase/FertiScan/delete_inspection_function.sql

```diff
@@ -0,0 +1,125 @@
+-- Function to delete organization_information
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.13".delete_organization_information(
+    p_organization_information_id uuid
+)
+RETURNS void
+LANGUAGE plpgsql
+AS $$
+BEGIN
+    DELETE FROM "fertiscan_0.0.13".organization_information
+    WHERE id = p_organization_information_id;
+END;
+$$;
+
+-- Trigger function to handle after organization_information delete for location deletion
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.13".after_org_info_delete_location_trig()
+RETURNS TRIGGER
+LANGUAGE plpgsql
+AS $$
+BEGIN
+    IF OLD.location_id IS NOT NULL THEN
+        DELETE FROM "fertiscan_0.0.13".location
+        WHERE id = OLD.location_id;
+    END IF;
+
+    RETURN NULL;
+END;
+$$;
+
+-- Trigger definition on organization_information table for location deletion
+DROP TRIGGER IF EXISTS after_organization_information_delete_location ON "fertiscan_0.0.13".organization_information;
+CREATE TRIGGER after_organization_information_delete_location
+AFTER DELETE ON "fertiscan_0.0.13".organization_information
+FOR EACH ROW
+EXECUTE FUNCTION "fertiscan_0.0.13".after_org_info_delete_location_trig();
+
+-- Function to delete an inspection and related data
```

Comment:
This whole functioon should be only a trigger. Also I dont get what `related data` this is deleting? The comment is missleading

Reply:
I was expecting this function to call and delete the label_information data

Reply:
yes, deleting the label_information is a solution but it is strange to delete a label_information when the function is named delete_inspection even though the inspection will eventually be cascade deleted.